### DISCUSSION TITLE
How can I determine the width of a text box that relies on fallback fonts?

### DISCUSSION THREAD

**Discussion by ekkxl** [prawnpdf/prawn#1280] [Q&A]
I'm trying to figure out the width a piece of text so I can do some layout before positioning the text.  The text includes words and symbols, and I need to use multiple fonts to cover all the needed glyphs.  The fallback fonts mechanism works very well for that.  However I'm having trouble figuring out how to determine the width of the text.

Document has width_of, which is giving me the wrong width.  Looking at the code for width_of, it doesn't appear to take fallback fonts into consideration.

Best I can tell, the only place that knows about fallback fonts is Prawn::Text::Formatted::Box, which I can create, and render with dry_run: true.  But there doesn't appear to be any way for me to get the width of text inside the box after that.

Is there someway to get the width of a piece of text with fallback fonts that I've missed?  I'm using the latest version, 2.4.0.  Thank you for any help,

**Comment by gettalong** [comment]
Have you tried the master branch? I remember seeing similar reports before and there is a change (https://github.com/prawnpdf/prawn/commit/73be039575a6e793235d03c4640da309e57b9ebf) that affects the width of text.

**Reply by ekkxl** [reply]
I tried the master branch, and width_of gives me the exact same results.  Looking at the code of width_of, it appears to measure the text in the current font, and never takes into consideration fallback fonts, even in the master branch.  Looking at arranger.rb, it doesn't have any logic for fallback fonts like box.rb does.

**Reply by ekkxl** [reply]
I was able to get the correct results by making a change to prawn.  First I added a line_widths attribute to Prawn::Text::Formatted::Box, then in Prawn::Text::Formatted::Wrap.print_line I added the following line at the end of the method:

@line_widths.push(  accumulated_width )

Then I can create a formatted box, render it with :dry_run, and read the line_widths.

Does that sound a like a reasonable change I could submit?
